File: 0137-single-number-ii/0137-single-number-ii.java
class Solution {
    public int singleNumber(int[] nums) {
        if(nums.length == 1) {
            return nums[0];
        }
        Arrays.sort(nums);
        if(nums[0] != nums[1]) {
            return nums[0];
        }
        else if(nums[nums.length - 1] != nums[nums.length - 2]) {
            return nums[nums.length - 1];
        }
        
        for(int i = 1; i < nums.length - 1; i += 3) {
            if(nums[i - 1] != nums[i]) {
                return nums[i - 1];
            }
        }
        return -1;
    }
}

File: 0137-single-number-ii/0137-single-number-ii.py
class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        hashmap = {}
        
        for i in nums:
            if i not in hashmap:
                hashmap[i] = 1
            else:
                hashmap[i] += 1
        
        for i in nums:
            if hashmap[i] != 3:
                return i